src/gen_context.c

```diff
@@ -45,23 +46,23 @@ int main(int argc, char **argv) {
     fprintf(fp, "#define _SECP256K1_ECMULT_STATIC_CONTEXT_\n");
     fprintf(fp, "#include \"src/group.h\"\n");
     fprintf(fp, "#define SC SECP256K1_GE_STORAGE_CONST\n");
-    fprintf(fp, "static const secp256k1_ge_storage secp256k1_ecmult_static_context[64][16] = {\n");
+    fprintf(fp, "static const secp256k1_ge_storage secp256k1_ecmult_static_context[ECMULT_GEN_PREC_N][ECMULT_GEN_PREC_G] = {\n");
```

Comment:
This is somewhat weird if you change the config in between but for some reason `gen_context` is not run again.

So I *think* the generated `ecmult_static_context.h` should have the `gen_context.c` compile-time values of `ECMULT_GEN_PREC_N`  and `ECMULT_GEN_PREC_G` hardcoded and contain a precompiler assertion (`#error`) that verifies they're identical to the `ecmult_static_context.h`-compile-time values. This should make sure that we never compile with a `ecmult_static_context.h` generated with a different configuration. (Alternatively, we could make the file name of `ecmult_static_context.h` make dependent on the configured values, e.g., `ecmult_static_context_64_16.h` or something.)

But maybe I make a mistake here, it's easy to get these things wrong. So correct me if you think I'm wrong.

Reply:
That's a great catch. Normally gen_context on `make` is run if the configuration changed, but it might not be absolutely failsafe (e.g. by manually overwriting the file with a different version etc.).

Change incoming.